com.unity.multiplayer.mlapi/Editor/PostProcessScene.cs

```diff
@@ -11,7 +11,7 @@ public class NetworkScenePostProcess : MonoBehaviour
         [PostProcessScene(int.MaxValue)]
         public static void ProcessScene()
         {
-            var traverseSortedObjects = FindObjectsOfType<NetworkObject>().ToList();
+            var traverseSortedObjects = FindObjectsOfType<NetworkObject>().Where(x => x.IsSceneObject == null).ToList();
```

Comment:
Could we get some comments on this method, what it is for, why it is here, and maybe a brief explanation on NetworkInstanceIds and how they relate to other types of NetworkObjects (i.e. dynamic and/or players)?

Reply:
Yes, but I think this PR is not the place for that.

This is a hotfix that should be merged ASAP.

But +1 for comments in the method. Will write that up

Reply:
The only thing I will say about this code is that it doesn't scale and I agree with @NoelStephensUnity that it would be good to understand what is actually going on here and how can we make it less suck later. `FindObjectsOfType` is a horrible function and then add to the fact you are filtering after that then ToList which is generating garbage.

Reply:
It scales well enough for a OnBuild method. That's the reason it uses Linq. Because it's not used in realtime code.

But that's besides the point, this is a fix and not a rework.

Reply:
The argument that well just because its editor code it can be slow is not a valid one :P

Reply:
That was not my argument. My argument was that it's never been shown to be slow enough to be a problem.

> It scales well enough for a OnBuild method.

Reply:
Yeah, given the time pressure I'm not sure today's the day to come up with a different scheme than having it scan for all the NetworkObject-enabled methods.

But I would like to capture the subtlety here, because the definition of IsSceneObject itself is a little non-intuitive, namely (from its definition in code) : "null if it's not yet spawned but is a scene object."

So please do add a comment saying "find all scene objects that have not been spawned yet", because when I read this code 'cold' I assumed the exact opposite.